src/Silex/CallbackResolver.php

```diff
@@ -33,35 +33,49 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            if (!isset($this->app[$service])) {
+                throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
+            }
+
+            return array($this->app[$service], $method);
+        }
+
+        if (!isset($this->app[$name])) {
```

Comment:
@HeahDude, ok, so convertCallback assumes that the name is valid. So doing the `isset` test on line 60 is redundant, so I will remove.

However, the check at line 53 is different because we aren't testing if `$name` is in the container, but if the service part is in. The `$service` would be the first part of the `$name` var.